src/pluggy/callers/cythonized.pyx

```diff
@@ -0,0 +1,64 @@
+"""
+Cynthonized hook call loop.
+
+This is currently maintained as a verbatim copy of
+``pluggy.callers._multicall()``.
+
+NOTE: In order to build this source you must have cython installed.
+"""
+import sys
+
+from .._result import _Result, _raise_wrapfail, HookCallError
+
+
+cpdef _c_multicall(list hook_impls, dict caller_kwargs, bint firstresult=False):
+    """Execute a call into multiple python functions/methods and return the
+    result(s).
+
+    ``caller_kwargs`` comes from _HookCaller.__call__().
+    """
+    __tracebackhide__ = True
+    results = []
+    excinfo = None
+    try:  # run impl and wrapper setup functions in a loop
+        teardowns = []
+        try:
+            for hook_impl in reversed(hook_impls):
```

Comment:
i wonder if there is a way to line up implementations better so we cna do less work and less work on calls,

Reply:
@RonnyPfannschmidt oh there definitely is. I actually have some ideas but it's going to require some rejigging of how implementations are stored/managed in the manger iirc and I think this may have something to do with the way we couple tracing to that management (see #262 and #217). I also think we should let @bluetech play with it in the context of `pytest` before we get too ahead of ourselves - micro-benchmarks shouldn't be our guiding light IMO.

I also thought about this a while back with ideas for the rework of internals.. let's see if I can find some notes.

@RonnyPfannschmidt another question I have is how to implement both of these implementations (cython and python) without too much duplicate work.

Reply:
@goodboy as cython has a python mode, where a annotated file is next to the implementation file, i beleive we can do it better these days (by implementing it in python, and then cythonizing those files

Reply:
So then it is as simple as running cython on the original code?

Reply:
im going to sprint on this next week, i hope to get a sync with simon from datasette about async use of pluggy before so we can enable that as well